public/app/features/trails/ActionTabs/MetricOverviewScene.tsx

```diff
@@ -91,26 +93,31 @@ export class MetricOverviewScene extends SceneObjectBase<MetricOverviewSceneStat
             <Stack direction="column" gap={0.5}>
               <Text weight={'medium'}>Labels</Text>
               {labelOptions.length === 0 && 'Unable to fetch labels.'}
-              {labelOptions.map((l) =>
-                getTrailFor(model).state.embedded ? (
-                  // Do not render as TextLink when in embedded mode, as any direct URL
-                  // manipulation will take the browser out out of the current page.
-                  <div key={l.label}>{l.label}</div>
-                ) : (
+              <MemoryRouter>
+                {
+                  // Using a memory router allows us to use TextLinks in embedded mode where there might not be a wrapping Router
+                  // Avoids "Invariant failed: You should not use <Link> outside a <Router>"
+                }
```

Comment:
There should always be a wrapping router, this was a bug in some our modal/drawer rendering 

Fixed this problem in https://github.com/grafana/grafana/pull/84916

Reply:
I made a note to remove the drawer after that PR is merged, in case this gets merged first.